Modify MuonDirectionCorrectedCaloHitEnergy histogram to enable correct calibration of CLIC/CLD muon detector system

oviazlo opened the pull request:
src/CalibrationHelper.cc:
- Change x-range of MuonDirectionCorrectedCaloHitEnergy histogram (from 0 - 1.0 to 0 - 0.002) in case when detector model has non-scintillator muon chambers (CLIC, CLD). Otherwise all entries will end up in a single bin which will lead to wrong value of MuonGeVToMIP calibration constant

calibration/PandoraPFACalibrate_MipResponse.cc:
- Remove initialization of histograms. Clone them from input file instead.

oviazlo commented:
@webermat Matthias, can you also have a look if it looks OK for you?

webermat commented:
As far as I checked an upper limit of 1./500. aka 0.002 with 2000 bins, which leads to a granularity of 1.e-6 is sufficient for CLIC needs, where the peak should sit around 5.e-5.

StevenGreen1 commented on calibration/PandoraPFACalibrate_MipResponse.cc:
```diff
@@ -226,13 +214,31 @@ void GeVToMIP::MakeHistograms()
 
                 if (i_hECalDirectionCorrectedCaloHitEnergy!=NULL)
-                    m_hECalDirectionCorrectedCaloHitEnergy->Add(i_hECalDirectionCorrectedCaloHitEnergy,1.0);
+                    if (m_hECalDirectionCorrectedCaloHitEnergy!=NULL)
```
Can you please format this in Pandora style of bracketing the if expression if it's followed by an else or else if.  Can you also do this for two following if statements too.

StevenGreen1 commented:
Hi Oleksandr, thank you for making this pull request.  I think the cleaner way of making this change would be to add a configurable parameter to define the maximum value of this histogram (and the number of bins if you feel that would be useful) to the pandora_analysis::CalibrationHelper::Settings class.  Then when the MuonDirectionCorrectedCaloHitEnergy histogram is created it will use whatever parameters you chose at runtime. This way you can change these settings whenever you like in the future without having to make a PR.  Also there are two histograms doing exactly the same job in this feature branch, which isn't ideal coding practise and so having one configurable option would be much nicer and a very welcome contribution if you agree.  What do you think?  I am happy to offer advice if anything is unclear with my explanation.

oviazlo commented:
Hello Steven. Thank you for comments! I changed the code according to your suggestions and add two configurable parameters to define the maximum value and the number of bins of MuonDirectionCorrectedCaloHitEnergy histogram. Also I fixed style of bracketing. Please have a look. Thanks!

StevenGreen1 commented:
Excellent work!  Thank you for doing this, I think it shows a great template of how people can make similar changes in the future too for the other calibration variables.  Apologies for my slow response merging this, it's been a busy week.  I will merge this now though.  Best Wishes, Steve
